### PR TITLE
Switch to using POSIX pseudoterminals.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
Previously, we were using SysV pseudo-terminals on Solaris and BSD
pseudo-terminals on other Unix platforms. BSD pseudo-terminals have
been deprecated on Linux and are no longer available in some kernel
configurations.

The POSIX API is basically the same as the SysV API, apart from using
`posix_openpt` instead of `open` with `/dev/ptmx`.

Closes interlisp/medley#121.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
@masinter Please give this a try and see if it works for you. (I've only built it, not tried running it with a sysout.)

**Review by nbriggs — APPROVED** [review]
LGTM.  Checked on macOS 10.11 and Solaris 11.4.

**Comment by nbriggs** [comment]
I'll wait for @masinter's approval before merging, or Larry can do it if he chooses.

**Pull request merged**
